TombEngine/Game/effects/lightning.cpp

```diff
@@ -26,9 +29,168 @@ namespace TEN::Effects::Lightning
 	float FloatSinCosTable[8192];
 	Vector3i LightningPos[6];
 	short LightningBuffer[1024];
-	
+		
 	std::vector<LIGHTNING_INFO> Lightning;
 
+	 TWOGUNINFO twogun[2];
+
+	 void TriggerLaserBeam(Vector3i src, Vector3i dest, short LeftRight)
+	 {
+		 TWOGUNINFO* tg;
+
+		 int j;
+		 for (int i = 0; i < 2; i++, i++)
+		 {
+			 tg = &twogun[LeftRight];
+			 for (j = 0; j < 4; j++)
+			 {
+				 if (tg->life == 0 || j == 3)
+					 break;
+			 }
+
+			 tg->pos1 = src;
+			 tg->pos4 = dest;
+			 tg->pos.Position = src;
+			 tg->life = 17;
+			 tg->spin = (GetRandomControl() & 31) << 11;
+			 tg->dlength = 4096;
+			 tg->r = 0;
+			 tg->b = 255;
+			 tg->g = 96;
+			 tg->fadein = 8;
+
+			 TriggerLightningGlow(tg->pos.Position.x, tg->pos.Position.y, tg->pos.Position.z, 64 + (GetRandomControl() & 3) << 24, 0, tg->g >> 1, tg->b >> 1);
+			 TriggerLightning(&src, &dest, 1, 0, tg->g, tg->b, 20, (LI_THININ | LI_THINOUT), 19, 5);	
+			 TriggerLightning(&src, &dest, 1, 110, 255, 250, 20, (LI_THININ | LI_THINOUT), 4, 5);
+		 }
+	 }
+
+	void UpdateTwogunLasers()
+	{
+		for (int i = 0; i < 2; i++)
+		{
+			TWOGUNINFO* tg = &twogun[i];
+			int	j;
+
+			for (j = 0; j < 4; j++, tg++)
+			{
+				if (tg->life)
+				{
+					tg->life--;
+
+					if (tg->life < 16)
+					{
+						tg->spinadd -= tg->spinadd / 8;
+						tg->size++;
+					}
+					else if (tg->life == 16)
+					{
+						tg->spinadd = MAX_VISIBILITY_DISTANCE;
+						tg->coil = MAX_VISIBILITY_DISTANCE;
+						tg->length = tg->dlength;
+						tg->size = 4;
+					}
+					else
+					{
+						tg->coil += (MAX_VISIBILITY_DISTANCE - tg->coil) / 8;
+						if (tg->dlength - tg->length <= (tg->dlength / 4))
+						{
+							tg->spinadd += (MAX_VISIBILITY_DISTANCE - tg->spinadd) / 8;
+							tg->length = tg->dlength;
+						}
+						else
+							tg->length += (tg->dlength - tg->length) >> 2;
+						if (tg->size < 4)
+							tg->size++;
+					}
+
+					if (tg->fadein < 8)
+						tg->fadein++;
+
+					tg->spin -= tg->spinadd;
+				}
+			}
+		}
+	}
+
+	void CurlSpline(Vector3i* pos, short* buffer, TWOGUNINFO* tg)
+	{
+		buffer[0] = pos[0].x;
+		buffer[1] = pos[0].y;
+		buffer[2] = pos[0].z;
+
+		buffer += 4;
+
+		int segments =  tg->segments ;
```

Comment:
Why do you need to make new variable with a copy of `tg->segments` value? From the further code it seems you are not modifying it, so you can use `tg->segments` everywhere.

Reply:
fixed